File: ToDoList/app/src/main/java/com/davdo/todolist/src/NoteDao.java
package com.davdo.todolist.src;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;
import java.util.UUID;

@Dao
public interface NoteDao {

    @Query("SELECT * FROM note")
    public List<Note> getNotes();

    @Query("SELECT * FROM note WHERE id=(:id)")
    public Note getNote(UUID id);
}

File: ToDoListKotlin/app/src/main/java/com/davdo/todolistkotlin/db/NoteDAO.kt
package com.davdo.todolistkotlin.db

import androidx.lifecycle.LiveData
import androidx.room.*
import java.util.*

@Dao
interface NoteDAO {

	@Query("SELECT * FROM note")
	fun getNotes():LiveData<List<Note>>

	@Query("SELECT * FROM note WHERE uuid=(:id)")
	fun getNote(id: UUID): LiveData<Note?>

	@Update
	fun updateNote(note: Note)

	@Insert
	fun addNote(note: Note)

	@Delete
	fun deleteNote(note: Note)
}